godot-core/src/builtin/quaternion.rs

```diff
@@ -130,70 +138,115 @@ impl Quaternion {
         Quaternion::new(v.x, v.y, v.z, 0.0)
     }
 
+    /// Normalizes the quaternion.
+    ///
+    /// # Panics
+    /// If the quaternion has length of 0.
     pub fn normalized(self) -> Self {
-        self / self.length()
+        let length = self.length();
+        assert!(!length.approx_eq(&0.0), "Quaternion has length 0");
```

Comment:
Note to self: we could maybe consider a method `ApproxEq::approx_zero(&self)`. But not in this PR 😉